inst/components/scss/sidebar.scss

```diff
@@ -371,7 +372,7 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_padding-icon)), var(--_sid
 
       // Soften main area when sidebar is expanded over content
       > .main {
-        opacity: 0.33;
+        opacity: 0.2;
```

Comment:
I'm a little confused as to how this works. I would've expected the main area to be 100% opacity at all times, and for the backdrop's opacity to determine how much the main area is visible?

Reply:
The latter opacity meaning this, I think: https://github.com/rstudio/bslib/blob/c5244cfae75c434e5fae6091820c13b62de53fd7/inst/components/scss/card.scss#L139

Reply:
We could go that way, but the current set up fades out the main area when the sidebar is expanded. I have some vague memories about why its set up this way:

1. It's part of the sidebar expanding animation to fade the main area while the sidebar is expanding (on mobile)
2. We don't actually have a backdrop element, we're just overlaying the sidebar over the main content area. It's technically possible for the sidebar not to be full-width, in which case the sidebar isn't fully a backdrop and it makes sense to de-emphasize the main area.
3. Varying the background color opacity is possible (we'd have to change opacity from 0.05 to something closer to 0.8). But that effect would only apply to the default background color and would break if users give semi-transparent background colors. The fact that background color has a few different paths for user configuration means we'd have to think about it more. I guess this point is that it was vaguely easier to modify main opacity staying out of the way of the more complicated user input paths.

Reply:
These all seem like reasons to do it the way I'm suggesting. In my mind, the physical metaphor is that the sidebar is a piece of white, translucent glass being slid over the top of the main content. You don't need to fade out the main content independent of the sidebar; what de-emphasizes it is the fact that there's a piece of glass sitting on "top".

[Current method](https://posit-dev.github.io/py-shiny/shinylive/py/editor/#code=NobwRAdghgtgpmAXGKAHVA6VBPMAaMAYwHsIAXOcpMASxlWICcyACAMyhpIgGcWo+HADoQ6DZi1QAbYmSnYMcAB6pGcHnwGSlIkQGIWAGWJQAJi1NQyUfhHMl6AVwose1slxYA3KFMfqRNkZiGFcACyg1czEmVjRUAH1TGkY8Fg9UHkDg0J4wmghsAHcaUwBzODI+GIk1OzhGBOlZeV0IIJDwguwWGtY1KEIPLzg0utMG7M687sUVNQ1e+ljeiFRnNMcaNoAjGikpBMYIMpYAXhYAChgCy4yeDDJZXwS9g4BKNJgoJTuaTMez0Obyk73ebQMAEFTOZUFAKukaHI4LZzDxSnAdpERFssPC4AliKgqn9kWchGAAErqayORhQciI9AFMoUtJsfZSKA7KRwM4AFUY-nBEBxNEe8IebmwvMuIhYCpYFOVYHliowOx4UhoOwAtOiJljGLqudhiM4AHwYA2YyIsEBqxWK3W6hI2o26mDEN5wT0-XUlUxkMKIFgAdgArABSADcjpYAF94xqtTr9RiPabzWRgJZrLqSAc0Oieb6vT7yWAyEK4BSALpW74Fe3xpMQJ0q5UQEVtErBli492RS5EyiViY8ADWT1QFPeiHjuIK6zIbu1E0YcvbTsVFKe1mBnLZ8Y7YAAQpz+F7HORj9udywbhAziCjidgAAGOt4E+K75KF9OTfMpgAARm-X8FR8Pw+VfY4yh-e8d1UPkKQAEjvJ0RSdJc1mcBJCDCOBCEnHZiCUBIymCRxUC3B8FT3Oha3wSClTAAAxYhiDRBovC4ZjEPolhgApQwb0Itk2IAEQKCBJjACCkKdHg4F5IY4FMM4RLAMSIAk-BpNk+TFKEgptTkgUa0ExVsMVXCVwSQYPFIV5nCeCA5TABZKkkilqRU1gOSkChGDnCEWGhcwm3bbgKFvCAAEkAGEAHkADkAGVzhbJCKUcFTQqQdgoAwLgXJ4Lwyk8-L5LSCk1DKRwuUKz54wpIpfF5MgKVDDhSu4N1Ks8jqDh8sBWtyog6TqQhsF1UxiCLQrepKsqIEGqqKQWpb0zKUVxushiwFU7VMhoLIir6taNs8k7-nRC6JrbEQ+zCAdxSzfDC0cGBeEuIKpDOdjfBU+d41e96MGg-xXnIy48mIIpCAEPlkvSjLtJqwq6zBpTdzAfkgSZVAGgutohIAAXGBo5hQjRWImNh0iBBIMhJxgeEuXGhKde4kisKAuetCISc-OtycVCHcWhgkyN+BGkZRs40cy7SRq6+tufoilIRGekEQyLtWKpygN1phYsjxhVGf4PX8VZ-4uYXK2d3MC4+bzQXbJ53omdMYW0DgMWWAtFgP2dn2H3Zwhsv9jIWAAegsQED1eI8XaEtgKRAaOMHgBknYwUCoyTVVRSQqXxRl2H5bCRHkZU5XUtVilCGmyhZvmxbms1iPtbAXWGnxFgQSNl2TfqRhzfUS2hJtqA7YqNODid1inTdpkeH56wubXxUaD9gPRa-EOw77yPFRBWOU5eEE87gAvvYv9h0JAEFEAwAAmNhS7aF6kTeriT6q5vq-U5oWBIgZgw8C0gANjSPAlgoFP443PiwSuGBkaMFMP9JqhweCEDUGOQUwo0E4XFFg0wCQiJmAaJ5fk90LACzCuXSm1Mp6ewSEEWAtYXY22sKWVeGdFRqDIHSds7CMBSQFgAcUYKUP4mRt5ex7BXABkNKG4IOG6QhcBiE1i1g+DBlDqEPw3JcQgXINAJHHLqNgvIlAsAAFb5Q8GwOasVKBkF1DsSoRQ9Htl8DQPauokRwBgDwAsXj5KGJ5hSQmB4R6Xi8HwQ2ZcL4YIYAwPWpJeSVkivwFghYmDeEiDQbkvJJLSEGOErxlYZxzjIZHFWGMKR3TOhdcWwiHz2XwvSZIxBXJkHcpzPePsKQEKsCFAii0mCYWfg+NKpBRhjJ5tpCAyzfJgBUjoAyEyvSTlqmxSw2AtkeHgPWQ6Cz94QHMnyEhKzuk7lUWw02DQmgyDkKcvhcAmaTOGQ0ZoZAhEX2KYwbKy5nDWmRgCxoYLd5PNEeI7Q0Kpl0NWbze6yiuZXIvgBPcLNR4sSeU6bA9T-jzIWWCs4Sy5K+yKbM8FZwLgUg2XJCkLBVIqQZTIVIGLFTVlNncysMh-H02JRfF5kt1G4k0WwPBOiiHPgebEp0xjIhUJoeYyxAgt62PscoZxriD4eNIHFbxviyD+MoPwbUISwkRKieQGJTSHzxP+JIBohAvH4jJk8zJRJiA5I8GSHWMJClgtKfIipzFVXNObq046Bx7rnXrPyhUvTVz9JoIMnYblSCjJJf3DITQvUJFORK65p4MrSCRCPbAC5K1VoVNpHZWyeAHKOVtKAFa6pViYpc9NO4VJqQoJpbtvah1OjMgUe5Vkh1SofBPDcHyWjfLnr8pkpbGCEBBZHDooR5HlDgECpY4h+ilAqECiWPs8zZQ9gLBFF88zaWjl4+sscrCPA9UnPMN9DwHFWdgHwTLVgrm-YkaO5an2R0cNBPgFwX3AciHWDAN4aAAEd-AItWTwWA0h1DZWALmL9b7yAkZzMhxgdZzgXGgnWGjbASnQVWAOeDXTVnSFYBcI9V7Pl0WuXh+gvIYFCYIzwXFkcuS+KkDAuDINJM+yxHYKBYQzgfgwB+UCimeZgsmXkikfEj2pqbc-MFXoJiVmCEUUJ9Rdk6awjenmXG0OoDzASYBAmFm8gqHYM4yQhheebUwGgPrnLPgpGESS2AGSESYJWMiwyQjRbOKBTTn80hKFi3XRglZvXOsKpl9TGAIwLtWYunmSLjiSGCn-CAuFLGOAmARDQlx4hJBSInNi0oROYPFaogwLohvDZG6Nsb42JuTZdPoFg1InI0BGEU3wbcuThc0HYTlbA2DESqDNqb+2DuHdG3Vk282RiYOWyIfhWLPZ7oVFfC4kKyD-uXlIGDCpShKFAve+6L276WutR5EEYs0jA-Ak-T7n8fsAnOXAUq6IPJPe-fALmT8qvtnuMAT732ABkvRTBKBQSdgYQwFtw9+dtoYIhTuk-O3AEY5BLhI+8mQEUNsEh3chjRdza4MSbgJanIlaQZaAW0fBJ+uJudWAJIRYipFyKUWorRRiFyDIjp2xpLSolxJRb2WAGSEA5LYxFGABMeBwDQHgNQNQWGUi1PIA8MgSgyD4CIGarx1AOBrUtjMQoIhffFEvZUS2QLWhrAZJYS2vGT2fJEKb83kAeHUDyJEDSWBcAEE8VQZAB7JBWDCNqHYZ6VgAAV89tD6HnuwWgtBubaO15IoGy-BkuAkLh+wCQJHeHiOo3VyCMIuG5jAAwqGEAqm19AHXwVJ0YgCMfXgWHx4t0n5APX1B9Z4K7rPLvkCIGCLIHKCoXSakL+mQ0kQ-QFB8WUUMeg2AAA4H8P7jBAZ6EAsCBr1oflgx-IlZKDQ0LqFqoATsDfiwHoJ-JAS-kfj4n-p-oAcAcaGCrfltmwC-m-h-tkg0CwBqGQBAAWDINyg6NuEFCFKGCBpcL-j4ngQQcQCpAGPkBQHYvsCFO8OgUvonlbsgPcH1l4Fvu7tnlWISpyHgBkHgDsuIZ2qkCcqIUxOITQAAF68KgSwIYAACcaheAaWWmeA7E4SvgowSyeAGUN4eABuRueAn8IgoEGmAAzAACxaEYCwLwIACyBheARhJhEAZhRkqQthIgn82h2mthJWeA7hvInhxAxhph5hDQeAAREAn8oRsC9+CRGAth2mERhh0R3hvhhu8RVhSR9hJWmhKR2meh3wkRXhsRfheA9hgREYX8mhJRYYWRHhNRPhcRqQDREA9+GAYYYYlhmm4RHRuRtRBRqQRRn8qh9+aRoRyCox1R4xXRdRvRoETRH4jhaWahbhYxMRqxkxlh1hrR8xX8thSxORBx+RFhRRNhX8QxaWbRlxUR1x3RxxEAthTRMxeAmxuh+hyxbxax1hTR9hGWaWEYQx2RrxeR7xdx-R9hFxoRH4LxnRNx8R6xJR9+SJ6W-xVRVxsJdR0xEJZxahGW0JaJcJ1hdhFxaWsCQxlR+xhJRxiRKhzR6RzxFJKx6J-h1hqhahQxoREYeJTJExFhiRn8GmsCwpoRXxqJuRVgPJCR1hYY6hGWJRH4aRXJxhipVJSRGmn8mhn8AxwpjJgJGUupRJIJAxQxxpyRIp5plpRxRRthahGAYJeAqpEYWpTJTpFh6xbp9+GWCxPpjpZASpdxqp9+2mxpDh8pOp4Z7x6xsptpIx2pFpiZVpEAyCzhRpaZvpmZzpgRTx9RGR8ZGZEZ1hbp0p6RKJZpBJfphRIgbpEYwpaW9hwp6ZjZUx1h-RcpxpnZBZSp6xmx8CyJDpDZhZtxgRUpeZHZE5MJ3Z9RKpAxZxn8UJQ5epn8GpY5zh5ZS5rJqhmRwxKJXZU58Rh56hOJH4DJAJk5SprJ-RsCeZx5Z5D5IgmRDxvx+ZYZ752ZqhWxwx25+555vJ2Zqp9h8CA5jh6ZYpTZ2ZoRahjhyJex5pcFPZEA1ZaRaWth5Jophx05nxGmjhTRqFBJ6Fy52Z-RhptZ5ZFF0xdpFxfxsFBF8FuF7peAAFdFrFPRgR-RrZJ5g5aFPFypVFmmyFP55FIldx0FwxQlUllZ2ZBpeZ8C9ZMJ9FH5WJ2m3xC5lJwJmF6h2xJWZF6l0ljRxlpZcZLFw5VZ7pZRJW7Rwlilth-ROh450JAAmuoAmTZUkaofYdphCbpQqaBR8aBGlsBU8SZV5RJhWXqfYdRUMaoWGBcW+UmTOeyaEUES8TFT5XqaBKEdGSeTld5XFVmeFbiTsaGaMLlWVUWWJTRaEWGNFaVUuesV6SpTBZuXURKRppqekRsSBX+WyfOcafYbefiYuaFS6Zpo8SMWpbVUuaBBlfacaToWleVZsTGZppoZ5a1dNdYWteCeoRub+flaqTebRWpV4QeYEbMbuaBI4QtftYpZ-N8WkX8c9bFUuXca0RcXacFXleVRphGIFV-PJVNYpaBKqWSbWdVXgItQdUkaqeqZJa8fyGEHSHgLpIRJRZAQMbuaefsRjVjTjWEGFSjXmZycTZjakGTWFW6fYRJUTZEblSTXTTrmFaoS4ekV1YCezdjZzXcX1XNdiYDQLfTctZ8caakd+SzTVd5RLULdaWoWkWtalTTaTcrZ8ViUxUXBNZrRzXpOTcmZpv9WjUYUrcbWFdRWkRqfGVbbjdMa0dpRgPfhDZbbTYLdbcSUXITQ7V7fTdMf0QKbRXtRJuxPIq9WNfZR7dEZHTQIpU0ZGE4XWXeQjd5QnVDUeajRcV9boVHVucaSnVlaaenUYVnfFRptDaWSlSVRHYXfpeuV-HbWjblZXeVerU4RDe3Y3fVUEZpiEW3ZnX3YRTmZBd3fGR3fVRVf2WEfndPWPU0S5Vdenb3YnfFSUebWXZNYjX+fjR6aEQlQuXvVuRprDRqWkddSFVDcvfAnSXzQrd9UjZKXZaWR+C1c-a9d8dpiUTReHUDTPdRRJc1UNflVvb-fNWvS9eAyMe5d1fVSjW2VA5NTdaFVLS5bNe-dxfvYhcKQOXHUCf3cnZA2nag9ye8ayaqceUKYDcyYRYaVeXLadQpelUpc4cgzvYbd7U7Yde6bSW7eLYHdrduZpmrWqUI1rdbesXSS+Y5VcY7eTXcUfcdbLdfXgIo2FX-VBWEdqZo3cU0WofgxbdEfo3w8fQORluo2Y0kYGaWVY+XaY8I9bbAiIPxcdcfXo847w9mSURGBlmtZI0bT47PeI2Q9w0HYEcaUGcMaBPDZ7VI7jayW6f1UFV44k0oytamWLdY945k9mWljXUFbkxk2FXaTo+E-zXkzbV-LuW9UEzw-k-xWDfUyU8E-k-cbYebZUwo9U79UXMYz0+jX03ySMYEwAzYwVX7cMSnW040xTe6YKfPY4xoyMzrWWclQHaU64xAAFUXJocxRE5zaVsjZpsKZ9Ss5ozsyNebXnZc2s32Zww05E2JbAkZbYapfc6U6yWlvfnNbAnc+Q6s6U0UW6WGNhQMRrVUyCyIF6dtfLcMzC741gwOTg-lQsd02i+VSjTo2GI-aZfvX44M1i0cQGTacVdZWw4w+7d+Y9XQxRScwlfrXLSSxYTs29XAyafSyJaba6cMbAhUSs-Q-BSNXmQi3paS8WWEbQ5S-pTmTzai7KzPaodlYE+o8K2BVM0VeM-hVDXC6vbvaVRpcUSVsGe6fExQz1RlaA3afixK4RZ+c8X46yxidSaUU4QKyfTA8Dc4Zi-nXVYRRGBpkVfcQi6fVa9mU0UVQsZ-YA4RUg7Sz3Ua2ZZ8b8zKUXPDbVca2yTE-bQAxqx8exfZeFfXUQ2PVGWK6WwW-0xY8PbFca9Wdg-68aw4b6+kQJfmy2yUTWSkZm8m75ckRkXrTWZ2ymx2Rkem0m-WyJRKQsUlbox4VmzyyIPYSDem2G-22fbmSy0qxYSc6-bHV69O3+bYcRRU1Wwy72QM+20e2WxeYEQsdg6O8NaRZoWk4u5u+VYGZA6BJofnTY5-AiY4QObG5M6CWE3hazYrWs2yd6U+x+xJjY16ffe6Xa040i508B3W8C+0x8YiUXK7UM2zdU6bSiWq9A4hzB0KfC6W5M-0U69h5o6ycXVhzof+1R2Mygwhzh-M3cf5XJbewByDRcXm9x5o+sf5Us-aex983de60ffIxnV-ZQx+fR9eVW21YEX-amR+FC0-XGyK8XTo9JxRwZxhae5pue-m7dUkVGwc9M9Z6Fb0Uy103gCk2A3K1lamVw1B8p1mYB8y6oT5-pwG-e9mcaW+0XLe6FxhQF4NfA752Z78dYRCZF1O0l0UaqXi04UzRp0jRZ3S8iX+6ZzF5RTmSoUBaW1nS81M4fea7e9V9rfxRC7DQA419bVLRseoWcUMxXVHfTaySo93X2w3YnTVyq95w1-19raG5WwvdNz7RlfOQl-eWw1M4BzM5B6t-58aYsdE1fSVz9Vp4PRS9x6VyOduyt0p0lz884UPXyxtTPRpoMTlyw5DflU0RGOp451DSLU4fU492PTLanXl85dEwIzXT93qV8W7SJ84QbWdRG4w7p9+a1wg+KYEaqXE8Vf60d0kTLTp6D+dW7cdSnYDyK-Rz1w03qabnWEAA)
[Suggested method](https://posit-dev.github.io/py-shiny/shinylive/py/editor/#code=NobwRAdghgtgpmAXGKAHVA6VBPMAaMAYwHsIAXOcpMASxlWICcyACAMyhpIgGcWo+HADoQ6DZi1QAbYmSnYMcAB6pGcHnwGSlIkQGIWAGWJQAJi1NQyUfhHMl6AVwose1slxYA3KFMfqRNkZiGFcACyg1czEmVjRUAH1TGkY8Fg9UHkDg0J4wmghsAHcaUwBzODI+GIk1OzhGBOlZeV0IIJDwguwWGtY1KEIPLzg0utMG7M687sUVNQ1e+ljeiFRnNMcaNoAjGikpBMYIMpYAXhYAChgCy4yeDDJZXwS9g4BKNJgoJTuaTMez0Obyk73ebQMAEFTOZUFAKukaHI4LZzDxSnAdpERFssPC4AliKgqn9kWchGAAErqayORhQciI9AFMoUtJsfZSKA7KRwM4AFUY-nBEBxNEe8IebmwvMuIhYCpYFOVYHliowOx4UhoOwAtOiJljGLqudhiM4AHwYA2YyIsEBqxWK3W6hI2o26mDEN5wT0-XUlUxkMKIFgAdgArABSADcjqdLBdboxHp2ZVDjDKWMuACYIxG0nmCywi2kAAwYAAc7zjEATztd7sinu9+19O2ISn1EVMxCKoYAjGXUEoWEOR2Ph6PM9my+X5ywK2WIzX4wBfeMarU6-Up5um81kYCWay6kgHNDonm+r0+8lgMhCuAUgC6Vu+BXt8adRMGSOwg61k6G51oqKrKhAIptCUwYsLiTaMJcRKUPeEw8AA1k8qAUu8iDxriBTrGQbrahMiHfmBD5Aq8nJshRCoUgAQpy-Beo45B0aB9Y3BAZwgkcJzAGWL54PRLDfEofGcgJZTAAOIliT4fh8vxxxlKJXEJqofIUgAJJxToik6BFrM4CSEGEcCEOhHZKAkZTBI4qByppToUh48AGfWSpgAAYsQxBog0XhcM++BicAFKGOxFlsj5AAiBQQJMYAKa5io8HAvJDHAphnJFYDRRAsX4AlSUpWl3mrNqyUCk+GmGfh4qEWZgweKQrzOE8EBymACyVHFFLUplrAclIFCMDhEIsNC5gfnW3AUBxEAAJIAMIAPIAHIAMrnF+mkUo4mWTUg7BQBgXAdTwXhlL1x0pWkFJqGUjhcqdnzxhSRS+LyZAUqGHCXdwbq3b1P0HANYCfYdRB0nUhDYLqvYXqdgMXVdECg3dFIo+9u5lKK0MNZRWXapkNBZGdQOY9jvVk-86JUzDIEiDBYRweKB5meejgwLwlxjVIZy+b4mW4fG7OcxgSn+K8naXHkfaEAIfLrdtO0FQ9p0vhL6UMWA-JAkyqANFTbRVQAAuMDRzNpGhiRMbDpNRGSm4wPCXHrVVOvcSRWFAXvWhEptCS+FuKlLuKywStmK2Eyuq2c6u7QVEN-a+3veRSkIjPSCIZBBYnW5QZF2wsWT6xYcDO1Aef4gkGRe3hVdOuYFx+yegdGT7Co0M7pjB2gcBhywFqLi3vf1u7hD7YPGQsAA9BYgLWMCtGt95bAUiAM8YPADLNxgA5RhuqqippUfijH8u-ErRQq5lyebanFKEPDlCI8jxCo5nk-Z2AXODR8QsBBEXKuJd6iMHLuoSuVUnb8HrhUGiBxm5iTbvtTuAcvboMVP3FeeRh6j3HmWf+U9FQgjnqvF4IJ95wEPj3chCpt5gF0iAEEiAMA5jYGfNobMkQc1xNzYivN+ae3PAkQMwYeD5QAGxpHkWOHMusyEsCvhgFWjBTCCzeocHghA1AoUFMKVRxlxSaNMAkSyZgGi9X5IzCwAcpoXytjbaBXcEhBFgM+KuCDrDXjQZvBUagyB0jrG4jA8UA4AHFGClD+Jkf21gvZQUvgI6WFidEHDdAYuARinxZ3rOoixVj6FkUuIQLkGgEioV1GwXko4ABWx0PBsCRotSgZBdQ7EqEUPJdZfA0EJrqJEcAYA8DPJ0lKhSfYUiNmvUBLEvB8ELufch6iGAMDzqSXk95Zr8BYOeJg3hIg0G5LyOK0hBhjM6feLCOFTFTxTprCkDMKZU3DkEhMJkiJHDMDQYgnUyDdU9rg3uFJ9FWAmuZH+TAvJMITFtUgowwU+wKhAZFg0wCZR0KVCFXp0KPR8pYbAWKPJhUqgihMBQap8mMSir5jUXHeUgWRJoMg5Ckt8TXVwKtgUNGaGQQJ5CjmMH2i1Mg1o+XQtFTgr5ISwnaClVC2xqLfaMySd3EmCLJLuWomA8KjLFTYDuf8eFTDRVnCRclXoztRXnAuBSDFyUKQsCyplQ5sLUhqsVI+UutL7wyD6Q7Q15DUlOmKZEbRbBdE5MMbxelMyI3pNxCU6x5TKkCB4DU3GdSGksGaW4fu7TSBLS6T0sgfTKD8G1MM0Z4zJnkGmY8+scz-iSAaIQTp+JzZfI2USYg2yPBkhzjCA59qfBxPOWFJNTyX4vLAG8pmr4fUKh+WZekyRAU7C6qQUFRrW0Pn+E0TtCRSWhqpW5MAO1pBIlAQBc1l6Co4qxTwAlRLcZQHPU9I9nlUrasva4LKVkKB5U-d+1d1KIABvpQBsNEd6ysoaOyloXL4E8oyCexghBhVTw6KEOJ5Q4CCqWOIfopQKiCoQz7E8mCNVdzleQk8BUZ6dNfHPKwjx23LxPNQ9eBxUXYEneK0ykrMMzzPYxqejglJ8AuMxoTkQXwYHYjQAAjv4OVqKeCwGkOofawBjycdY+QIzR5FOMBfA6k5UgXxWbYMcpSqw4Kyc+ai6QrALiEcoxylyl6dP0F5DIgLemeBwanlyHpUgZEybFuF3uWI7BSLCGcJcA54s+1FZC3ZFIQqEcpo+qeoqvQTHvMEIoIz6i4oy0y9z40VOoBPASYRfmqW8gqHYM4yQhitcAywJgNBu3tV4hSMIcVsAMgskwe8HZgUhHG2cAcGAyw5jSEoSbCdGD3i7U206a3UsYAjJBxhPtw1TwVccSQ40+EQBMpUxwExzIaEuPEJIKQl4+WlEFjRIbUkGBdADwHQPgcg9B2D8HLp9AsGpG1GgIxDm+HflyYbmg7BurYGwEDld-sQ9x3j-HQObsl1hyMDRiORB+Po9gmZlCLgSr4ygqQUm+6mCUAOOjAInhr0ZxqXp-TLggjDmkQX8kTulCUDmDnDxyWXXRD1en5KUnxgu3We4wBxfs4AGS9FZ8oonAwhhw7gIoDHIGRDE8N6TuAIxyCXHp-1MgIoEEJFw2u8UTkmskQxORKi3ODVpBjlJbJakTu4g91YAkFkrI2U7PZRyzl3J0DCmkTK2VQP5SijFMbeKwCJQgMlHWIowBrjwOAaA8BqBqA0ykG55BpdKDIPgIgpbOnUA4JjSuMxCgiC78UCjlRK6CtaGsBklhK7eeIxykQxfS+QG8dQQhUQsC4AIB0qgyB8OSCsGEbUOxSMrAAArb7aH0LfdgtBaEa20V7yQxUXCP8GS4CRPFtmf+8PEdR-rkAcRcRrGABiWKEA3QvboBvZirLyJ4AhAFeDOIz5l7z7IBfbqA-Y8BN5r6N7ICIDBCyAHQKguiai767iGjNjzTdLpgsB6BsCVhUFUG1iswQBYADp5y4GJjdITKbKDoNC6jppcFpihh6A5iCFASsGai6gcF5zcFlJcGir8EY5sB0FtCMFbINAsAahkAQBngyAeoOigRjQTShiTqXD4ETI7DqGaHECZQBj5AUB1L7ATSrgQBrhwFz4V7ID3A-ZeBoEt7r6+40Kch4AZB4A4pBHvqpAkoBFJ5BE0AABePiA4siGAAAnIkXgEtmWOlr5GMr4KMEingDtOxHgHngXngDmCIEOBgAAMwAAsqRGAsi8iAAstkXgLkfkRAIUeVKkBUSIDmGkelhUYdngE0byC0cQHkQUUUQ0HgN0RADmAMbIpWNMZUelsMTkWMW0R0fnlMaUbMVUYdikfMRkVkSMa0RMZ0XgFUT0RGFwikXsWGCsc0ace0ZMakJcRAJWBgGGGGCUctkMY8esWcVsakDsTmAkZWIsQMQOKtqsaMeMc8ecW8QONcWWDUUtokY0f8XCZscUTsQOHcRCVwhUX8ScQCfCUCSUWURWDmN8UtvccSWsViS8RSRABUdcaCXgMiXgJkd8CSYyQiWUdcVUatkthGN8TCU8didsWUR8VUUSQMXOOKaSZKa8WUXsZWHKctqttyZiRsUySCSKQSYkdCTqYCTiZSZUUSUtrIt8dqbybqecTMfETcUsXSYqXyeSY6QkYkd8QMQWLaQyfaR6T0RWLIgWAMayfSbCTtFYMqdMWUWGEkatnsWWIsW6dGWQLGSCVSSkTmJ8X6ccQGTGXqQKZ8d8bmXMVyQWVGUWecTsRUYkRgEKXgAmRGKmTqTWeSYiQ2ZWKtpCW2XaR2WaRAAOAmZWOlrmdUZGa0YOVMYieGWWb8WmTOcCWUbmbIjmYue2RmcWbMbSRcZUVOescucyQOA2aGUsXOP6dWdubWSIA2fmLUVUQWEuTeeSbiR8RGbmc+VubGYicifIvKZWTyYWa+UOTmCGRuU+UBT+UyYiQmWGASdSYeXkceSCcmQBXUchembGZ6csT8QqTBQ6WUV6RqWWDaVWdOaBVMY6R8euT8RUQ8QOVRV0SIAxVwt8ZyS+ThcRctjUeWTUVxUyY6QmVUfIl+QJSaWSUOQOAMYkTUfKRiXaaaVKRAGeYsUthUcaUpVJSpRURWDUdcYpQGcpSqcOR8TmAcZudpZmT0eWUSZxZJTZSybmTUQkQRdZbBT0R8Q+bmcuFhSZXGWZbxRef5TpSucOeJT8d+R5becOdmT8fIleRKTudUVWOlmydBTFZ2XeUkaiYdkZVGQFSCdcRGPInsZOWmQFV2Y2ZZRGIxcZWFcyRUR8ekSFTCQAJrqAoXMUXE9EJFVHpYimZUgVOUDhLY5h5VhgFWdVhbYU7lVHmXfEJFhhEmCX8mzHxUDG9GRkzXdWjUDFjn4U7VdVzWxVjWam1GJH9mjC7WnVvnSnOkDFTXHWzXHlwX7EJUSVMXcUbXLYElIlYXHm4VQUuXkXAXXlOXyk0m-FXm3XHkDjBmEn4X1UQ07lInLbjnLYpEdUnWoWrkXVolimEX3XDkJlkUhVJVHk9UzGglVgYV4nDV4Bw09XFVcKLGcmw240s2qmfFEnlmM2UWjUVh1W1F5iA3c2k2JkXnXVM1c1OXUlcL7nuUMn8hhB0h4BFQWS9WzG5nPWAVumq3q2a1hAnkJkWU-GumYmG2pDG0nkNlVHyVWU3VdXW0a1Z4nkJH1FLFfUq1q023u24lLjQ3qkC1jGu220I3OV1Hs1O2y1hbh0B0llXX4WrVW1+1u3FQm11lqn2XHxg1p1G3u1znLZ82x25EJ2Z0nnmWLHJnIUV1a1oWfHpVVjRW+2F2V36nHwYXK2wn11Z1eVJE+mx27W+RxLy0uW1XIWj00BOXXGRi1GXlVkj1j1o0JGaX7lEmc1hbT3j15lLFi2U1ckr2xVVEVgjn7krUvVH0z2eWzFm013D1dU71o2+WWmDE43b3H0k29EY1tXNHL031nUuXyJDVunP1nUaUFhfmM0AOjXXHNUU1L1P1f1DkO0HnQNb17VCW2WNm9mNmLGYN3VgUVhGlK0EMUVU1wOVEgN1E+1x1YOxXgU1VK3TVy07l5hcLpZ7Hm0f0MMk0DjmWO3PVrX8PlVcMw1IOvUS14m-H63E1Dlm0FhpGh18PFGR3NXLbfG12VWNU02yVQOHat2FWNWs10m12H2BnFHCV4W+kqOWMqUWVJG53k06O-nmnnlDWH190nkuVv3kPg3l3p222N0pk-GkNeNBNF1lFWkbkMV12RMd1lEDFNlWn+MF3+2JPDncNiXv3pMZ0N1J0GM92BPt0FMRX4NRVakUN4DeMgndn7lVMBNh0JNa2yIiDeXCkVMG0tP91ZOHara+UqO1PROVGLGDMROlO9OCFVgDPHwy0lMZNa2OkNmhOgN5PBOI1IW5kh0TOLO9PnXn2ePVPDPlPsnjPHM9NV1cIYUH0XOTPMneWDVcL5lNM1OXOB0WlHW7P5P7N7EA3nOvMnNOlziDO8NAuQk5Pz3fMbOS1VFD0vPrOJ0snlV4DLXxP3NtMQADXHwpEOW8neNHZ33LYFgc13N7Oos8Wyn0VDPvPtOVFKODHQtIsCO0O1EVGJVks-OBXnWVjQ2yKb2cswsNkIW1GX3dP3M7EtmY3FPNMSs83k3QOuNo2Qml0yvunSVm2Qt0PJVEW7GGNHVKvrWnmlkGuOW32OOVgkvHw1EWMBWEsLV50cll1KlMmYscNzhPUItZXFHF31kJVHGvP2PhVOlVEblqtBva0-1hm5Peuzn41e2KtmtnUJHbUAtJtBnDkHXSuhWjVSuIPg23VFU9F7ERh4MLU5vYO-XPX8UVuxVsVmMxsNVuNxUfVWkBv-1sNnUhmqswOdsk0RgViHXlE93M0-Xo2HUQtX1EMqWKNOsA28MRt1lLaWtLECNTtFvDlglJmP2zUbuaVJGrso2FsmPxlVhhvrsnt9Plugsdu7uXtnlK29t3uQ17H8tLEPkLt7uvvRsLEXvNtzGfOGVGPHtOVPn0vvt-uVsAfn0JHAcnVVUiCn36tyMjEgfsMQVOvht2uI2T2EMBUzF6WNk5MjvweNWEsssA22N4e6M9GQmPufs0fDmGUpFrOoekejXdniOnlPtvNyuzEyl8X5VX1AuCljMXVgu0ubuHYP0kfx2Sctk0MO3ovksfOTk3tsdyd8eynHzN2L0Fsu2XPF0gsSP6eacqdJOHbZsSd8cssNuye8fkuOm62CetWYPgu-FpsacOdcu4n9VRU8cnNMMRhEnaO3vee22In9VD0VlueXM02weWUDVTtA2sUfGX0ofO1SP-vcMLllip1efTvBu605MxeSOqO6VLjEfJfU1XFntzusNZe30OsVFEkrPi3NsyXsX+eEN438eOuwc8eFfcu5ksfHyDe9c5gfGUc7vlepDkcimjdweNexXwV5VKcLu9eEcM3ykpE9c9WIm5nxE-FaWZfX0+cWdNnJNpNec70wveXqVS28O3fu2R3o3J2AWH3PeV2OnJOdOtlX1fdlNOlbOMtleA-7NLjnuYPg-MlMNQUZeo1GtbVidhgneI-f2HezPgnjcS0TW-3qendDd-l1GWX2dDeOlttLF+siPSUVhfGPlE3fVo0lWkXVdC2aOi1GOC0v2YXKMbcS37sh1Lbn38+Q3XEh2vv51M+6uON5dOukM0-UU9EJlrueeE8Tdrm5ds9o2jmdPz2K-BtpcEl6d5M7nF4vhAA)

(I also added a slight drop shadow to further the physical metaphor but I think the point holds without it)

For the third point (users specifying more- or less-transparent sidebar bg colors), if they're specifying a more transparent sidebar bg, wouldn't that be because they want what's behind to show through--and vice versa for less transparent?

Reply:
I also recognize that we're essentially out of time here. We might need to negotiate the release schedule with @icarusz on Monday; I know Carson still has at least one or two high priority issues to fix as well.

Reply:
> In my mind, the physical metaphor is that the sidebar is a piece of white, translucent glass being slid over the top of the main content.

This is a great way to think of it, and I agree that it's generally the right metaphor. The problem is that the sidebar background is at the intersection of a few different design considerations that make this difficult. 

The short summary is that the sidebar is only incidentally white in this one context -- a default sidebar layout in the Shiny preset without any local color adjustments.

One primary design consideration is that the sidebar background default needs to work over any color in both light and dark mode. Our default uses a highly-transparent foreground color -- `rgba(var(--bs-emphasis-color-rgb, 0,0,0), 0.05)` -- to slightly lighten or darken the main color below the sidebar.
   
![image](https://github.com/rstudio/bslib/assets/5420529/10653037-c501-49e8-b7d2-36b6125ed6a5)

That means that we can't set the sidebar background to a low-transparency color by default. We'd instead need to vary only the opacity component, e.g. making the sidebar less transparent when it expands. But this approach won't work in the above cases because we'd be moving from white text on a dark background to white text on an almost white background.

Reply:
OK, thanks for the explanation. I'll approve this PR as is.

Reply:
FYI, I spent some time investigating using `color-mix()` here and unfortunately, that doesn't help us either. The problem is that we can't write CSS rules that reliably know the background color over which the sidebar element is overlaid. Unlike `color`, which cascades and has a `currentColor` property value, we can't query `background-color` in order to put together the `color-mix()` declaration. Without that, we're just using a very high-powered hammer to apply transparency.

The examples above demonstrate the problem. We have the following markup (sketch):

```html
<div class="card bg-text-primary">
  <div class="bslib-sidebar-layout">
    <div class="main"></div>
    <aside class="sidebar"></aside>
  </div>
</div>
```

Here the primary colored background is coming from the `.bg-text-primary` class and `.main` has `background-color: transparent` by default. This means that we can't write rules for `.sidebar` that know about the background color of the card unless we modify the `.bg-text-primary` class so that it sets an intermediate CSS variable, e.g. `--bs-card-bg` or `--bslib-bg` property. IIRC, we also encountered this problem with the `value_box()` CSS and decided against getting into modifying these Bootstrap-provided classes.

When the background color is driven by the sidebar `.main` area, specifically via `layout_sidebar(bg=)` or by using the appropriate CSS variables, then we _do_ have the background area in a CSS variable and could use `color-mix()`. 

I just updated the PR to use `color-mix()`, but I had to resort to using CSS selectors like `[style*="--_main-bg"]` to differentiate between layouts that have a main/sidebar bg set and those that don't. This leaves us using the "fading `.main`" approach is as few places as possible.